# HG changeset patch
# User Andriy Shinkarchuck <adriano32.gnu@gmail.com>
# Date 1310841326 18000
#      Sat Jul 16 13:35:26 2011 -0500
# Node ID 7b90b81a5793675f34c6e2b97b5f72ce858a472a
# Parent  a85ddaf8f402b3153bfe91f2b9c6ec32484588c9
codesprint: Wrote 8 tests for sinewave.m

diff --git a/scripts/signal/sinewave.m b/scripts/signal/sinewave.m
--- a/scripts/signal/sinewave.m
+++ b/scripts/signal/sinewave.m
@@ -38,8 +38,18 @@ function x = sinewave (m, n, d)
       n = m;
     endif
     x = sin (((1 : m) + d - 1) * 2 * pi / n);
   else
     print_usage ();
   endif
 
 endfunction
+
+%!assert (sinewave (1), 0);
+%!assert (sinewave (1, 4, 1), 1);
+%!assert (sinewave (1, 12, 1), 1/2, 1e-6);
+%!assert (sinewave (1, 12, 2), sqrt (3)/2, 1e-6);
+%!assert (sinewave (1, 20, 1), (sqrt (5)-1)/4, 1e-6);
+%!assert (sinewave (1), sinewave (1, 1,0));
+%!assert (sinewave (3, 4), sinewave(3, 4, 0));
+
+%!error sinewave ();
